### PR TITLE
openring: init at f13edb5d

### PR DESCRIPTION
**Pull request by sumnerevans**
###### Motivation for this change

Initialize the openring package from https://git.sr.ht/~sircmpwn/openring

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
   - Tested in a Nix shell: https://git.sr.ht/~sumner/sumnerevans.com/tree/master/item/shell.nix
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/openring/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ buildGoModule, fetchFromSourcehut, lib }:
+buildGoModule rec {
+  pname = "openring";
+  version = "master";
```
Comment (sumnerevans) [inline pkgs/applications/misc/openring/default.nix:4]: Should I use something different for the version? Maybe this?

```suggestion
  version = "unstable-2021-04-03";
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118414` at 0335bde4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>openring</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/openring/default.nix:20:5:
```
   |
20 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/openring/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/openring/default.nix:4]: Yes please.

File: pkgs/applications/misc/openring/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ buildGoModule, fetchFromSourcehut, lib }:
+buildGoModule rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openring/default.nix:2]:
```suggestion
{ buildGoModule, fetchFromSourcehut, lib }:

buildGoModule rec {
```

File: pkgs/applications/misc/openring/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromSourcehut {
+    owner = "~sircmpwn";
+    repo = pname;
+    rev = "f13edb5dfd882ce608d61cf6b6740650ce9d84a3";
+    sha256 = "sha256-Z65V77JZ9jCzBg7T2+d5Agxxd+MV2R7nYcLedYP5eOE=";
+  };
+
+  vendorSha256 = "sha256-BbBTmkGyLrIWphXC+dBaHaVzHuXRZ+4N/Jt2k3nF7Z4=";
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openring/default.nix:15]: Why are tests disabled?

File: pkgs/applications/misc/openring/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description = "A webring for static site generators";
+    homepage = "https://git.sr.ht/~sircmpwn/openring";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ sumnerevans ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openring/default.nix:22]:
```suggestion
```

set by buildGoModule

File: pkgs/applications/misc/openring/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  vendorSha256 = "sha256-BbBTmkGyLrIWphXC+dBaHaVzHuXRZ+4N/Jt2k3nF7Z4=";
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A webring for static site generators";
+    homepage = "https://git.sr.ht/~sircmpwn/openring";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/openring/default.nix:20]: gl3Only or gpl3Plus?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/openring/default.nix (hunk 5 above)
Comment (sumnerevans) [reply pkgs/applications/misc/openring/default.nix:20]: I've asked upstream developer on IRC to clarify.

File: pkgs/applications/misc/openring/default.nix (hunk 3 above)
Comment (sumnerevans) [reply pkgs/applications/misc/openring/default.nix:15]: Two reasons, (1) I copied from a different package that had this set to false, and (2) there aren't any tests in the upstream package.

Comment (sumnerevans) [reply pkgs/applications/misc/openring/default.nix:16]: Should I just remove this, even though there are no tests?

Comment (SuperSandro2000) [reply pkgs/applications/misc/openring/default.nix:16]: Adding a comment that the package has not tests is fine.

File: pkgs/applications/misc/openring/default.nix (hunk 5 above)
Comment (sumnerevans) [reply pkgs/applications/misc/openring/default.nix:21]: Relevant mailing list post: https://lists.sr.ht/~sircmpwn/public-inbox/%3C20210403182413.g64oc5alrgkbkmeh%40coruscant%3E

Comment (sumnerevans) [reply pkgs/applications/misc/openring/default.nix]: The maintainer clarified that it is `gpl3Only`. https://lists.sr.ht/~sircmpwn/public-inbox/%3C20210403182413.g64oc5alrgkbkmeh%40coruscant%3E#%3CCAEDVDIBZ5PX.1YXHCZQG995MC@taiga%3E

**Pull request merged**